### PR TITLE
Bump the message header version from 2 to 3

### PR DESCRIPTION
**Pull request by justinlin-linkedin**
Change the message header version from 2 to 3 so that we can save lifeVersion in the log.

### CODE REVIEW

File: ambry-replication/src/test/java/com.github.ambry.replication/InMemoryStore.java [hunk 1]
```diff
@@ -264,7 +264,7 @@ public void updateTtl(List<MessageInfo> infos) throws StoreException {
         }
         MessageFormatInputStream stream =
             new TtlUpdateMessageFormatInputStream(info.getStoreKey(), info.getAccountId(), info.getContainerId(),
-                info.getExpirationTimeInMs(), info.getOperationTimeMs());
+                info.getExpirationTimeInMs(), info.getOperationTimeMs(), info.getLifeVersion());
```
Comment (zzmao) [inline ambry-replication/src/test/java/com.github.ambry.replication/InMemoryStore.java]: Just want to make sure one thing. For V2 and V2, info.getLifeVersion() returns -1, right?

Comment (justinlin-linkedin) [reply ambry-replication/src/test/java/com.github.ambry.replication/InMemoryStore.java]: it returns 0 here. 0 should be the lifeVersion to be persisted in the log.

File: ambry-store/src/main/java/com.github.ambry.store/BlobStore.java [hunk 2]
```diff
@@ -507,14 +507,9 @@ public void delete(List<MessageInfo> infosToDelete) throws StoreException {
         List<MessageInfo> updatedInfos = new ArrayList<>(infosToDelete.size());
         int ind = 0;
```
Comment (jsjtzyy) [inline ambry-store/src/main/java/com.github.ambry.store/BlobStore.java]: nit: could you change `ind` to `index`?

File: ambry-store/src/main/java/com.github.ambry.store/BlobStore.java [hunk 3]
```diff
@@ -619,16 +614,9 @@ public void updateTtl(List<MessageInfo> infosToUpdate) throws StoreException {
         List<MessageInfo> updatedInfos = new ArrayList<>(infosToUpdate.size());
         int ind = 0;
```
Comment (jsjtzyy) [inline ambry-store/src/main/java/com.github.ambry.store/BlobStore.java]: same here

File: ambry-tools/src/main/java/com.github.ambry/store/DumpDataHelper.java [hunk 4]
```diff
@@ -92,6 +92,20 @@ static LogBlobRecordInfo readSingleRecordFromLog(RandomAccessFile randomAccessFi
                 + header.getUserMetadataRecordRelativeOffset() + " dataRelativeOffset "
                 + header.getBlobRecordRelativeOffset() + " crc " + header.getCrc();
         totalRecordSize += header.getMessageSize() + buffer.capacity();
+      } else if (version == MessageFormatRecord.Message_Header_Version_V3) {
+        ByteBuffer buffer = ByteBuffer.allocate(MessageFormatRecord.MessageHeader_Format_V3.getHeaderSize());
+        buffer.putShort(version);
+        randomAccessFile.read(buffer.array(), 2, buffer.capacity() - 2);
```
Comment (jsjtzyy) [inline ambry-tools/src/main/java/com.github.ambry/store/DumpDataHelper.java]: Could you explain "- 2" here?

File: ambry-store/src/main/java/com.github.ambry.store/BlobStore.java (hunk 2 above)
Comment (justinlin-linkedin) [reply ambry-store/src/main/java/com.github.ambry.store/BlobStore.java]: we already have a member variable named "index", it's persistentIndex in the blobstore. I guess I can just name this variable to "i" if you desire.

File: ambry-store/src/main/java/com.github.ambry.store/BlobStore.java (hunk 3 above)
Comment (justinlin-linkedin) [reply ambry-store/src/main/java/com.github.ambry.store/BlobStore.java]: above.

File: ambry-tools/src/main/java/com.github.ambry/store/DumpDataHelper.java (hunk 4 above)
Comment (justinlin-linkedin) [reply ambry-tools/src/main/java/com.github.ambry/store/DumpDataHelper.java]: 2 is the number of bytes to store header version (it's a short). I copy this code from case above.

File: ambry-store/src/main/java/com.github.ambry.store/BlobStore.java (hunk 2 above)
Comment (jsjtzyy) [reply ambry-store/src/main/java/com.github.ambry.store/BlobStore.java]: yeah, "i" make less ambiguity.

**Pull request merged**